name.abuchen.portfolio.tests/src/name/abuchen/portfolio/datatransfer/pdf/merkurprivatbank/MerkurPrivatBankPDFExtractorTest.java

```diff
@@ -0,0 +1,75 @@
+
+import static org.hamcrest.CoreMatchers.is;
+import static org.hamcrest.MatcherAssert.assertThat;
+import static org.hamcrest.collection.IsEmptyCollection.empty;
+import static org.junit.Assert.assertNull;
+
+import java.time.LocalDateTime;
+import java.util.ArrayList;
+import java.util.List;
+
+import org.junit.Test;
+
+import name.abuchen.portfolio.datatransfer.Extractor.BuySellEntryItem;
+import name.abuchen.portfolio.datatransfer.Extractor.Item;
+import name.abuchen.portfolio.datatransfer.Extractor.SecurityItem;
+import name.abuchen.portfolio.datatransfer.actions.AssertImportActions;
+import name.abuchen.portfolio.datatransfer.pdf.MerkurPrivatBankPDFExtractor;
+import name.abuchen.portfolio.datatransfer.pdf.PDFInputFile;
+import name.abuchen.portfolio.model.BuySellEntry;
+import name.abuchen.portfolio.model.Client;
+import name.abuchen.portfolio.model.PortfolioTransaction;
+import name.abuchen.portfolio.model.Security;
+import name.abuchen.portfolio.model.Transaction.Unit;
+import name.abuchen.portfolio.money.CurrencyUnit;
+import name.abuchen.portfolio.money.Money;
+import name.abuchen.portfolio.money.Values;
+
+@SuppressWarnings("nls")
+public class MerkurPrivatBankPDFExtractorTest
+{
+    @Test
+    public void testWertpapierKauf01()
+    {
+        MerkurPrivatBankPDFExtractor extractor = new MerkurPrivatBankPDFExtractor(new Client());
+
+        List<Exception> errors = new ArrayList<>();
+        
+        List<Item> results = extractor.extract(PDFInputFile.loadTestCase(getClass(), "Kauf01.txt"), errors);
+
+        assertThat(errors, empty());
+        assertThat(results.size(), is(2));
+        
+        new AssertImportActions().check(results, CurrencyUnit.EUR);
+        
+        // check security identification:
+        Security security = results.stream().filter(SecurityItem.class::isInstance).findFirst()
+                        .orElseThrow(IllegalArgumentException::new).getSecurity();
+        assertThat(security.getIsin(), is("IE00BJ0KDQ92"));
+        assertThat(security.getWkn(), is("A1XB5U"));
+        assertNull(security.getTickerSymbol());
+        assertThat(security.getName(), is("XTR.(IE) - MSCI WORLD REGISTERED SHARES 1C O.N."));
+        assertThat(security.getCurrencyCode(), is(CurrencyUnit.EUR));
+
+        // check buy transaction
+        BuySellEntry entry = (BuySellEntry) results.stream().filter(BuySellEntryItem.class::isInstance).findFirst()
+                        .orElseThrow(IllegalArgumentException::new).getSubject();
+        
+        assertThat(entry.getPortfolioTransaction().getType(), is(PortfolioTransaction.Type.BUY));
+        assertThat(entry.getPortfolioTransaction().getDateTime(), is(LocalDateTime.parse("2023-05-02T09:34:40")));
+        assertThat(entry.getPortfolioTransaction().getShares(), is(Values.Share.factorize(125.3258)));
+        assertThat(entry.getSource(), is("Kauf01.txt"));
+        assertThat(entry.getNote(), is("Ihr ETF-Sparplan Nr.      1"));
```

Comment:
To remove all the blanks between the text and the "`1`", add `replaceMultipleBlanks()` from the "`TextUtil.java`".

See --> [here](https://github.com/buchen/portfolio/blob/21334c1e4668b2809170f35c986b3023ae797bb2/name.abuchen.portfolio/src/name/abuchen/portfolio/util/TextUtil.java#L136-L140) <-- 

change to
`.assign((t, v) -> t.setNote(trim(replaceMultipleBlanks(v.get("note")))))`

and add to imports
`import static name.abuchen.portfolio.util.TextUtil.replaceMultipleBlanks;`

Reply:
In my opinion, this note is rather unnecessary.
It would be better if they exchanged the following line against

`Block firstRelevantLine = new Block("^Depotnummer [\d]+.*$");`

You can put the much more important order number into the note. `Auftragsnummer 284722/61.00`

Reply:
Good suggestion! I changed the option logic. I thought it would be nice to see both. I will see when further documents are available, what makes more sense.